manifests/charts/istio-control/istio-discovery/files/waypoint.yaml

```diff
@@ -182,13 +182,10 @@ spec:
         - name: ISTIO_META_MESH_ID
           value: "{{ (valueOrDefault .MeshConfig.TrustDomain .Values.global.trustDomain) }}"
         {{- end }}
+        {{- if .Values.global.proxy.resources }}
```

Comment:
IIRC, it's used for sidecar, should we introduce a new one for waypoint, because normally they should be different.

Reply:
@zirain thanks for the review. I asked this very question and [here](https://github.com/istio/istio/issues/51496#issuecomment-2159516841) is the response.

Reply:
I am fine having it be the same for now. more granular config is better handled by implementing option 2 in the linked issue anyway, I think.

Reply:
There is no "for now" - this is an API. Once someone start deploying istio and expecting the proxy resource to be used for waypoints ( and of course somehow not apply to sidecars if they use both) - we'll be stuck. 

Helm values are the install API, and it should be treated as such, it's not an experimental or accidental API. 

If you want something 'for now' - maybe an env variable, but even that has become another de-facto API.